benchmark/single-source/Substring.swift

```diff
@@ -267,6 +268,38 @@ public func run_SubstringComparable(_ N: Int) {
   CheckResults(count == N*500)
 }
 
+extension Character {
+  fileprivate var isASCIIWhitespace: Bool {
+    return self == " " || self == "\t" || self == "\r" || self == "\n" || self == "\r\n"
+  }
+}
+
+extension Substring {
+  fileprivate func trimWhitespace() -> Substring {
+    var me = self
+    while me.first?.isASCIIWhitespace == .some(true) {
+      me = me.dropFirst()
+    }
+    while me.last?.isASCIIWhitespace == .some(true) {
+      me = me.dropLast()
+    }
+    return me
+  }
+}
+
+let _trimmableSubstrings = "pineapple,🍍,  pineapple\t,\r\n\r\n\r\n, 🍍 ,".split(separator: ",")
+
+@inline(never)
+public func run_SubstringTrimmingASCIIWhitespace(_ N: Int) {
+  let substrings = _trimmableSubstrings // bringing this alias from above
+  var count = 0
```

Comment:
Note to future self for a follow-up PR: Use `count` here to store the cumulative UTF-8 count of the substrings, then check the result at the end. That shouldn't distort the numbers at all and will enforce parity with alternate formulations.